### PR TITLE
Update CSV reader

### LINKED ISSUE #5: Update CSV reader
1. Abstract a function that takes row => player
2. Allow read csv to take in a second parameter Dict[str, int] representing the title of each column (right now the best name I can think of is just `columns`
    1. Validate this against the first row
3. In the function that takes row => player, use this dict to read the correct column number (instead of hardcoding column numbers)
    1. eg `playerUsername=row[columns['username']] instead of `playerUsername=row[0]`

### PR DESCRIPTION
**Pull request by sheehui**
1. Created a config file to store the csv `column ref: index` dictionary
2. Abstracted a function that converts a csv row to a `Player`
3. Regarding the validation for the dictionary, I'm not too sure how it can be done. Since the data is stored in a config file, the 'key' would definitely exist, unless of course tampered with. Direct comparison with column name and config key seems a bit too strict and troublesome as any change in csv column name would trigger a change in `config.yml` and `angel.py`. I'm not really sure what is considered good practice here, but I feel that perhaps the user/editor of the config file should know they can only change the 'value' not 'key'. 

Any advice on point 3 + maybe other validation that is needed?

Update: I decided to add error handling for incompatible keys and bad indexes [@c94ee53](https://github.com/Tembusu-Student-Developer-Club/archangel/pull/9/commits/c94ee538d0fb026b3747cf387617bcf4cbefb952)

### CODE REVIEW

File: angel.py [hunk 1]
```diff
@@ -27,10 +27,16 @@
...
 import datetime
-# # FROMS
+import yaml
+import os
+
+# FROMS
 from models import Player
 from arrange import angel_mortal_arrange
 
+# Instantiate Logger
```
Comment (nicktohzyu) [inline angel.py]: Could you please implement this fix in a standalone branch/PR! No need to remove it from this branch tho

File: angel.py [hunk 2]
```diff
@@ -66,41 +68,53 @@ def read_csv(filename):
...
+        print(f'Processed {line_count} players.')
+        logger.info(f'Processed {line_count} players.')
         logger.info(f'person_list has been processed successfully')
     return person_list
 
 
-
+def convert_to_player(row):
+    config_file = open("config.yml", 'r')
+    index_dict = yaml.full_load(config_file)["player_attribute_index"]
```
Comment (nicktohzyu) [inline angel.py]: should this be wrapped in a try block too? Eg to handle cases where the config file is not present

File: angel.py (hunk 1 above)
Comment (nicktohzyu) [reply angel.py]: This section of code seems to be duplicated. Could we abstract a logger script that does this? The singleton pattern may be applicable for this usage (so multiple scripts reference the same logger object)

**Review by nicktohzyu — COMMENTED** [review]
Indentation fix was definitely a needed one, though it would have been better to do so as a separate PR. I'm unsure of what to do now, am thinking of just leaving it in this

File: angel.py (hunk 1 above)
Comment (sheehui) [reply angel.py]: Will do this in a new PR

File: angel.py (hunk 2 above)
Comment (sheehui) [reply angel.py]: Done in [feb8480](https://github.com/Tembusu-Student-Developer-Club/archangel/pull/9/commits/feb8480ad18e917bc8267478474c61e03bab2eb6)

**Review by nicktohzyu — COMMENTED** [review]
Tested and approved, just awaiting update to requirements.txt for new package requirement!

**Pull request merged**